opencsp/app/sofast/lib/ProcessSofastFixed.py

```diff
@@ -113,45 +96,37 @@ def calculate_mask(self) -> ndarray:
         ]
         mask = ip.calc_mask_raw(images, *params)
 
-        if self.params.mask.keep_largest_area:
+        if (self.optic_type == 'multi') and self.params.mask.keep_largest_area:
+            lt.warn(
+                '"keep_largest_area" mask processing option cannot be used '
+                'for multifacet ensembles. This will be turned off.'
+            )
+            self.params.mask.keep_largest_area = False
+        elif self.params.mask.keep_largest_area:
             mask = ip.keep_largest_mask_area(mask)
 
         return mask
 
-    def generate_geometry(self, blob_index: BlobIndex, mask_raw: np.ndarray) -> dict:
-        """Generates blob dataset from sofast dataset.
+    def load_measurement_data(self, measurement: MeasurementSofastFixed) -> None:
+        """Saves measurement data in class
 
         Parameters
         ----------
-        blob_index : BlobIndex
-            BlobIndex object with all blobs assigned
-        mask_raw : ndarray
-            Mask array
-
-        Returns
-        -------
-        dict
-            Key word argument dictionary for SlopeSolver (does not include "surface_data")
-
-        Sets Attributes
-        ---------------
-        self.data_geometry_general
-        self.data_image_proccessing_general
-        self.data_geometry_facet
-        self.data_image_processing_facet
-        self.data_error
+        measurement: MeasurementSofastFixed
+            Fixed pattern measurement object
         """
-        pts_image, pts_index_xy = blob_index.get_data()
+        self.measurement = measurement
 
-        # Process optic geometry
+    def _process_optic_singlefacet_geometry(self, blob_index: BlobIndex, mask_raw: np.ndarray) -> dict:
+        # Process optic geometry (find mask corners, etc.)
         (
             self.data_geometry_general,
-            self.data_image_proccessing_general,
-            self.data_geometry_facet,
-            self.data_image_processing_facet,
+            self.data_image_processing_general,
+            self.data_geometry_facet,  # list
```

Comment:
thanks for noting the type, consider noting it via self documenting variables such as `data_geometry_facet_list`

Reply:
Noted. I like that change!
That will likely be a more involved change. Maybe something for a new PR.